lib/chutney/TorNet.py

```diff
@@ -764,6 +764,19 @@ def _genRouterKey(self):
             sys.exit(1)
         self._env['fingerprint'] = fingerprint
 
+
+    def _setEd25519Id(self):
+        """Make chutney check for relay microdescriptors before verifying"""
+        datadir = self._env['dir']
+        key_directory =  os.path.join(datadir, 'keys', "ed25519_master_id_public_key")
+        with open(key_directory ,'rb') as f:
+            f.seek(32)
+            rest_file = f.read()
+            EncodedValue = base64.b64encode(rest_file)
+            ed25519_id = str(EncodedValue)[2:-2]
```

Comment:
Why do you ignore the first 2 and the last 2 base64 characters?

You can add a comment, or use a more specific function, like replace("=", "").

Reply:
I ignored the first two character because I am getting output in this format `b'GmQIJTZeoYTf7kQhYON73x/CmnSLM8ZTvqNYLH6YmCk=`, so I am removing the `'` and `=` from back and `'` and  `b` from front. That's why I was slicing the two characters.

Reply:
That's not how python strings work. The `b'` and `'` show you where the string starts and ends, and that it is a byte string.

When you do `str(EncodedValue)`, the byte string is turned into a unicode string, including the `b'` and `'`. Which is not what we want.

You can find out more about Python's byte and unicode strings here:
https://docs.python.org/3/howto/unicode.html

If you do:
```
EncodedValue.decode('utf-8')
```
The string will become a unicode string, and the `b` will disappear.

So let's do:
```suggestion
            ed25519_id = EncodedValue.decode('utf-8').replace('=', '')
```

Reply:
Thank you, will do so.

Reply:
Finally, let's check the length of the string. All ed25519-ids should be 43 characters long. If they are not, we should log a warning and exit.

You can see similar code here:
https://github.com/torproject/chutney/pull/61/files#diff-5fe4ecd0da6560d24d20e8ce90281abfR762

But we can check length using `len()`, we don't need to use regular expressions (`re.match()`).